Tests/tests/test_ScreenCapture.cpp

```diff
@@ -0,0 +1,164 @@
+    
+    EXPECT_CALL(frameBufferApiImplMock, fbCreate(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext** fbctx) {
+                *fbctx = &fbcontext;
+                return ErrNone;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbInit(::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx, VncServerFramebufferAPI* server, void* serverctx) {
+                if (nullptr != server)
+                {
+                    server->framebufferUpdateReady(&fbcontext);
+                    server->framebufferDetailsChanged(&fbcontext, nullptr, 1280, 720, 5120, nullptr);
+                    server->paletteChanged(&fbcontext, nullptr);
+                    server->logMsg(&fbcontext, "msg");
+                }
+                return ErrNone;
+            }));
+
+    ON_CALL(frameBufferApiImplMock, fbGetPixelFormat(::testing::_))
+        .WillByDefault(
+            ::testing::Return(&pixelFormat));
+
+    ON_CALL(frameBufferApiImplMock, fbGetWidth(::testing::_))
+        .WillByDefault(
+            ::testing::Return(1280));
+
+    ON_CALL(frameBufferApiImplMock, fbGetHeight(::testing::_))
+        .WillByDefault(
+            ::testing::Return(720));
+
+    ON_CALL(frameBufferApiImplMock, fbGetStride(::testing::_))
+        .WillByDefault(
+            ::testing::Return(5120));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbGetFramebuffer(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx) {
+                return frameBuffer;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbDestroy(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx) {
+            	return ErrNone;
+            }));
+            
+    EXPECT_CALL(service, Submit(::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](const uint32_t, const Core::ProxyType<Core::JSON::IElement>& json) {
+                string text;
+                EXPECT_TRUE(json->ToString(text));
+
+                EXPECT_EQ(text, string(_T("{"
+                                          "\"jsonrpc\":\"2.0\","
+                                          "\"method\":\"org.rdk.ScreenCapture.uploadComplete\","
+                                          "\"params\":"
+                                          "{\"status\":false,\"message\":\"Upload Failed: 6:'Couldn't resolve host name'\",\"call_guid\":\"\"}"
+                                          "}")));
+                uploadComplete.SetEvent();
+
+                return Core::ERROR_NONE;
+            }));
+
+    handler.Subscribe(0, _T("uploadComplete"), _T("org.rdk.ScreenCapture"), message);
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("uploadScreenCapture"), _T("{\"url\":\"http://non-existent-host.com\"}"), response));
```

Comment:
instead of non-existent-host please create a test socket that will accept and check the data. Example: https://github.com/rdkcentral/rdkservices/blob/ce27b599ed8fd7ba2247cdc154f2af0e935d54b7/Tests/tests/test_DeviceDiagnostics.cpp#L47